src/PostList/PostList.js

```diff
@@ -0,0 +1,99 @@
+import React, { useState } from 'react';
+import getDataFromUrl from '../api/getDataFromUrl';
+import postsUrl from '../api/postsUrl';
+import usersUrl from '../api/usersUrl';
+import commentsUrl from '../api/commentsUrl';
+import Post from './Post/Post';
+
+function PostList() {
+  const [preparedPosts, setPreparedPosts] = useState([]);
+  const [isLoadingPosts, setIsLoadingPosts] = useState(false);
+  const [isLoadedPosts, setIsLoadedPosts] = useState(false);
+  const [searchQuery, setSearchQuery] = useState('');
+
+  const getPreparedPostsFromServer = async() => {
+    const [posts, users, comments] = await Promise.all(
+      [getDataFromUrl(postsUrl),
+        getDataFromUrl(usersUrl),
+        getDataFromUrl(commentsUrl)]
+    );
+
+    return posts.map(
+      post => ({
+        ...post,
+        author: users.find(
+          user => user.id === post.userId
+        ),
+        comments: comments.filter(
+          comment => comment.postId === post.id
+        ),
+      })
+    );
+  };
+
+  const handleInputChange = (text) => {
+    setSearchQuery(text.trim().toLowerCase());
+  };
+
+  const debouncer = (func, time) => {
+    let timer;
+
+    return (event) => {
```

Comment:
він не приймає event
```suggestion
    return (...args) => {
```

Reply:
чому не приймає, все ж працює ?

Reply:
debounce це функція декорато, вона не має знати якого типу аргументи мають передаватися функції яка всередині. Я маю мати можливість передавати туди будь-що  а не тільки event